front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx

```diff
@@ -56,24 +54,17 @@ export default function AssistantBuilderDataSourceModal({
       title="Manage data sources selection"
     >
       <div className="w-full pt-12">
-        {orderDatasourceViewByImportance(dataSourceViews).map(
-          (dataSourceView) => (
-            <DataSourceViewSelector
-              key={dataSourceView.sId}
-              owner={owner}
-              selectionConfiguration={
-                selectionConfigurations[dataSourceView.sId] ??
-                defaultSelectionConfiguration(dataSourceView)
-              }
-              setSelectionConfigurations={(
-                func: SetStateAction<DataSourceViewSelectionConfigurations>
-              ) => {
-                setHasChanged(true);
-                setSelectionConfigurations(func);
-              }}
-            />
-          )
-        )}
+        <DataSourceViewsSelector
+          owner={owner}
+          dataSourceViews={dataSourceViews}
+          selectionConfigurations={selectionConfigurations}
+          setSelectionConfigurations={(
+            func: SetStateAction<DataSourceViewSelectionConfigurations>
+          ) => {
+            setHasChanged(true);
+            setSelectionConfigurations(func);
+          }}
```

Comment:
We should avoid anonymous function here as it will create unnecessary re-renderings of the underlying component.